languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java

```diff
@@ -48,33 +48,30 @@ public String getDescription() {
     return "'dhá' nó 'beirt'";
   }
 
-  private String[] getRestOfNP(AnalyzedTokenReadings[] tokens, int start, int end) {
-    List<String> ret = new ArrayList<>();
-    for (int i = start; i <= end; i++) {
-      ret.add(tokens[i].getToken());
-    }
-    return ret.toArray(new String[ret.size()]);
-  }
-
   @Override
   public RuleMatch[] match(AnalyzedSentence sentence) {
     List<RuleMatch> ruleMatches = new ArrayList<>();
     AnalyzedTokenReadings[] tokens = sentence.getTokensWithoutWhitespace();
-    int markEnd = 1;
+    int markDeag = 1;
     int prevTokenIndex = 0;
     String replacement = null;
     String msg = null;
     for (int i = 1; i < tokens.length; i++) {  // ignoring token 0, i.e., SENT_START
       if (isNumber(tokens[i]) && (i < tokens.length - 2 && isPerson(tokens[i + 1]))) {
-        markEnd = i + 1;
         if ("dhá".equalsIgnoreCase(tokens[i].getToken()) && (i < tokens.length - 2)) {
           for (int j = i + 2; j < tokens.length; j++) {
             if ("déag".equalsIgnoreCase(tokens[j].getToken())) {
-              markEnd = j;
+              markDeag = j;
               replacement = "dháréag";
-              String[] msgTokens = getRestOfNP(tokens, i + 1, j - 1);
-              msg = "Ba chóir duit <suggestion>" + replacement + " " +
-                String.join(" ", msgTokens) + "</suggestion> a scríobh";
+              msg = "Ba chóir duit <suggestion>" + replacement + "</suggestion> a scríobh";
+              RuleMatch match = new RuleMatch(
+                this, sentence, tokens[prevTokenIndex].getStartPos(), tokens[prevTokenIndex].getEndPos(), msg, "Uimhir phearsanta");
```

Comment:
For the `dhá` token to be correctly underlines you need to replace previousToken by `previousToken+1`. This will also highlight the other numerals properly (e.g. trí et al.).

Reply:
Right, doh